src/vendor.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2023 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::*;
+use crate::function::*;
+
+/// Functions for the Reset extension
+#[derive(Copy, Clone, Debug)]
+pub enum VendorFunction {
+    /// Sends a Vendor-specific message
+    Vendor {
+        eid: u32,
+        fid: u32,
+        arg0: u64,
+        arg1: u64,
+        arg2: u64,
+        arg3: u64,
+        arg4: u64,
+        arg5: u64,
+    },
+}
+
+impl VendorFunction {
+    /// Attempts to parse `Self` from the passed in `a0-a7`.
+    pub(crate) fn from_regs(args: &[u64]) -> Result<Self> {
+        use VendorFunction::*;
+
+        Ok(match args[8] {
+            0 => Vendor {
```

Comment:
This adapts the pattern from the Reset handler - but I've not driven this in anger yet to see how ergonomic it is.

Reply:
what is the 'args[8]' used in the match case here? args has been only 8 long an represented a0-7 in existing callers for other functions.

Reply:
Ah, that's a typo.